ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -672,24 +643,63 @@ public boolean getEntriesSince(StoreKey key, FindEntriesCondition findEntriesCon
       } else {
         logger.error("IndexSegment : " + indexFile.getAbsolutePath() + " index not found for key " + key);
       }
-    } else {
+    } else if (key == null || index.containsKey(key)) {
```

Comment:
No - but it was a bug that I discovered while writing tests and seemed easy enough to fix. Can revert if you think it is not required.